File: app/admin/router/system/supervisionItem.go
package system

import (
	"github.com/gin-gonic/gin"
	"haocean/health-enforcement/app/admin/api/system"
	"haocean/health-enforcement/app/core/utils/jwt"
)

// InitSupervisionItem 初始化监管事项路由
func InitSupervisionItem(e *gin.Engine) {
	v := e.Group("system")
	{
		// 公开路由
		v.GET("/supervision/category/list", system.ListSupervisionCategory)
		v.GET("/supervision/tree", system.GetSupervisionTree)
		v.GET("/supervision/children/:parentId", system.GetSupervisionChildren)

		auth := v.Group("")
		auth.Use(jwt.JWTAuthMiddleware())
		{
			// 监管事项 CRUD
			auth.GET("/supervision/list", system.ListSupervisionItem)
			auth.GET("/supervision/:id", system.GetSupervisionItem)
			auth.POST("/supervision", system.AddSupervisionItem)
			auth.PUT("/supervision", system.UpdateSupervisionItem)
			auth.DELETE("/supervision/:ids", system.DeleteSupervisionItem)
		}
	}
}

File: frontend/Haocean-Mobile/api/supervisionItem.js
import request from '@/utils/request'

// 查询监管事项分类列表
export function listSupervisionCategory() {
  return request({
    url: '/system/supervision/category/list',
    method: 'get'
  })
}

// 查询监管事项树
export function getSupervisionTree() {
  return request({
    url: '/system/supervision/tree',
    method: 'get'
  })
}

// 查询监管事项子项列表
export function getSupervisionChildren(parentId) {
  return request({
    url: '/system/supervision/children/' + parentId,
    method: 'get'
  })
}

// 查询监管事项详情
export function getSupervisionItem(itemId) {
  return request({
    url: '/system/supervision/' + itemId,
    method: 'get'
  })
}

// 查询监管事项列表（分页）
export function listSupervisionItem(query) {
  return request({
    url: '/system/supervision/list',
    method: 'get',
    params: query
  })
}
